controllers/tykoasapidefinition_controller.go

```diff
@@ -138,10 +154,25 @@ func (r *TykOasApiDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.
 		return ctrl.Result{}, nil
 	}
 
+	if tykOAS.Status.IsVersionedAPI {
+		err := r.handleVersionedTykOASAPI(ctx, log, tykOAS)
+		if err != nil {
+			log.Error(err, "failed to handle versioned Tyk OAS API")
+
+			return ctrl.Result{}, err
+		}
+
+		return ctrl.Result{}, nil
+	}
+
 	_, err = util.CreateOrUpdate(ctx, r.Client, tykOAS, func() error {
 		if !tykOAS.ObjectMeta.DeletionTimestamp.IsZero() {
 			markForDeletion = true
 
+			if err := r.updateBaseOasAPIVersionsStatus(ctx, log, tykOAS); err != nil {
```

Comment:
As far as i am concerned, this method only runs during the reconciliation to update the status of base API, right?. If that's the case, i'd suggest renaming the method name because the method must only update the status only during deletion. So, it'd be better to rename it like `reconcileDeletingBase` sth sth.